### PR TITLE
Java: tolerate fields and classes with the name "nil"

### PR DESCRIPTION
**Pull request by smowton**
fieldref_exprt confused ID_nil with ID_empty_string and therefore rejected "class nil { int nil; }" for example.

### CODE REVIEW

File: jbmc/src/java_bytecode/java_bytecode_parse_tree.h [hunk 1]
```diff
@@ -352,7 +352,8 @@ class fieldref_exprt : public exprt
 template <>
 inline bool can_cast_expr<fieldref_exprt>(const exprt &base)
 {
-  return base.get(ID_class) != ID_nil && base.get(ID_component_name) != ID_nil;
+  return base.get(ID_class) != ID_empty_string &&
```
Comment (tautschnig) [inline jbmc/src/java_bytecode/java_bytecode_parse_tree.h]: How about `return !base.get(ID_class).empty() && !base.get(ID_component_name).empty();`?

And while I'm at it: would mind adding a test case?

Comment (smowton) [reply jbmc/src/java_bytecode/java_bytecode_parse_tree.h]: Done

**Review by allredj — COMMENTED** [review]
:warning:
This PR failed Diffblue compatibility checks (cbmc commit: 796f7e15251be6ba356e54c92d3764e118033d92).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/103234960
Status will be re-evaluated on next push.
Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

  - the compatibility was already broken by an earlier merge.

**Review by owen-mc-diffblue — APPROVED** [review]

**Review by tautschnig — APPROVED** [review]
Thanks a lot for adding a test!

**Review by peterschrammel — APPROVED** [review]

**Pull request merged**